client/processdamage.lua

```diff
@@ -0,0 +1,102 @@
+local function isLegDamaged(injury)
+    return (injury.part == 'LLEG' and injury.severity > 1) or (injury.part == 'RLEG' and injury.severity > 1) or (injury.part == 'LFOOT' and injury.severity > 2) or (injury.part == 'RFOOT' and injury.severity > 2)
+end
+
+local function makePedFall(ped)
+    ShakeGameplayCam('SMALL_EXPLOSION_SHAKE', 0.08) -- change this float to increase/decrease camera shake
+    SetPedToRagdollWithFall(ped, 1500, 2000, 1, GetEntityForwardVector(ped), 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0)
+end
+
+local function chancePedFalls(ped)
+    if IsPedRagdoll(ped) or not IsPedOnFoot(ped) then return end
+    local chance = (IsPedRunning(ped) or IsPedSprinting(ped)) and Config.LegInjuryChance.Running or Config.LegInjuryChance.Walking
+    local rand = math.random(100)
+    if rand <= chance then
+        makePedFall(ped)
+    end
+end
+
+local function isLeftArmDamaged(injury)
+    return (injury.part == 'LARM' and injury.severity > 1) or (injury.part == 'LHAND' and injury.severity > 1) or (injury.part == 'LFINGER' and injury.severity > 2)
+end
+
+local function isArmDamaged(injury)
+    return isLeftArmDamaged(injury) or (injury.part == 'RARM' and injury.severity > 1) or (injury.part == 'RHAND' and injury.severity > 1) or (injury.part == 'RFINGER' and injury.severity > 2)
+end
+
+local function disableArms(ped, injury)
+    local disableTimer = 15
+    local isLeftArm = isLeftArmDamaged(injury)
+    while disableTimer > 0 do
+        if IsPedInAnyVehicle(ped, true) then
+            DisableControlAction(0, 63, true) -- veh turn left
+        end
+
+        local playerId = PlayerId()
+        if IsPlayerFreeAiming(playerId) then
+            if isLeftArm then
+                DisablePlayerFiring(playerId, true) -- Disable weapon firing
+            else
+                DisableControlAction(0, 25, true) -- Disable weapon aiming
+            end
+        end
+
+        disableTimer -= 1
+        Wait(1)
+    end
+end
+
+local function isHeadDamaged(injury)
+    return injury.part == 'HEAD' and injury.severity > 2
+end
+
+local function playBrainDamageEffectAndRagdoll(ped)
+    SetFlash(0, 0, 100, 10000, 100)
+
+    DoScreenFadeOut(100)
+    while not IsScreenFadedOut() do
+        Wait(0)
+    end
+
+    if not IsPedRagdoll(ped) and IsPedOnFoot(ped) and not IsPedSwimming(ped) then
+        ShakeGameplayCam('SMALL_EXPLOSION_SHAKE', 0.08) -- change this float to increase/decrease camera shake
+        SetPedToRagdoll(ped, 5000, 1, 2)
+    end
+
+    Wait(5000)
+    DoScreenFadeIn(250)
+end
+
+---applies disabling status effects based on injuries to specific body parts
+---@param ped any
+function processDamage(ped)
```

Comment:
Global functions hold the Pascal Case, not camelCase, camelCase is used for local variables and functions

Reply:
While I understand this is a convention in the broader Lua world, since globals are resource scoped it is more useful to differentiate natives vs non-natives by grouping resource scoped functions (local and global) together.

We have a tradeoff with naming convention. We have local & global & native. So the two options are:
1. (local & global) & native
2. local & (global & native)

I'd posit option 1 is more useful than option 2. If you insist, I'll change it to option 1.

Reply:
I prefer option 2 myself, because anything global should be Pascal Case in my opinion no matter the scope

Reply:
done